modules/api/src/main/java/org/apache/ignite/internal/sql/ResultSetImpl.java

```diff
@@ -83,7 +85,7 @@ public void close() {
     @Override
     public boolean hasNext() {
         if (it == null) {
-            throw new SqlException("There are no results");
+            throw new SqlException(QUERY_NO_RESULT_SET_ERR, "There are no results");
```

Comment:
This is mostly a question to SQL guys, but are we sure we want to throw exception in this case at all? Why won't we just return `false`? How throwing an exception can be helpful for a user in this case? What do you think?

Reply:
We should throw an exception. Returning `false` will produce misleading behavior:

```java
        ResultSet rs = ses.execute(null, "INSERT INTO TEST VALUES (?, ?)", 20, 30);
        rs.forEachRemaining(System.out::println);
```

This will work without any errors if we return `false` from `hasNext`, but iterating over DML/DDL result set does not make sense. I think it's better to let the user know instead of silently allowing this.

Reply:
👍